### ISSUE TITLE
Can only pull from one array?

### ISSUE THREAD

**Issue by JulianKingman** [jagi/meteor-astronomy#190]
I have a funny scenario happening where I have two arrays in a Source that I'm trying update every time I remove a document from the Fragment collection, and only one array gets updated. The fullText array will update every time, but no matter what I do, fragmentIds refuses to update.

I have a schema like this:
````javascript
Source = Astro.Class({
    name: 'Source',
    collection: Sources,
    fields: {
        title: 'string',
        edition: 'string',
        fullText: 'array',
        link: 'string',
        book: 'string',
        sourceGroup: 'string',
        complete: {
            type: 'boolean',
            default: false
        },
        fragmentIds: {
            type: 'array',
            default: function(){return [];}
        },
        lockingUser: 'string'
    },
});
````

I then have this script that runs when a fragment is removed:
````javascript
Sources.find({fragmentIds: {$in: [fragmentId]}}).forEach(function (source) {
                source.pull('fragmentIds', fragmentId);
                source.pull('fullText', fragment.text);
                Meteor.call('saveSource', source);
            });
````

I can use set or pull, the result is the same, and if I only use fragmentIds, the result is still the same. The only differences I can identify between the fields are that 1. The first stores _id values, where the second stores text phrases, and 2. fragmentIds has a default value. Let me know what you think...

**Comment by jagi** [comment]
Yep, there seems to be an error. I will prepare a fix.

**Comment by jagi** [comment]
Ok, fixed in 1.2.2

**Comment by JulianKingman** [comment]
Awesome, thanks.
